src/hooks/useBanner.ts

```diff
@@ -0,0 +1,61 @@
+import { useMemo } from 'react';
+import dayjs from 'dayjs';
+import DOMPurify from 'dompurify';
+import yaml from 'js-yaml';
+import { useK8sWatchResource } from '~/k8s';
+import { ConfigMap } from '~/types/configmap';
+import { bannerConfigYupSchema } from '~/utils/validation-utils';
+import { ConfigMapGroupVersionKind, ConfigMapModel } from './../models/config-map';
+
+export type BannerType = 'info' | 'warning' | 'danger';
+export type BannerConfig = {
+  enable: boolean;
+  summary: string;
+  type: BannerType;
+  startTime?: string;
+  endTime?: string;
+};
+
+const parseBanner = (yamlContent: string): BannerConfig | null => {
+  const parsed = yaml.load(yamlContent, { schema: yaml.FAILSAFE_SCHEMA });
+  if (typeof parsed !== 'object' || parsed === null) return null;
+  try {
+    const banner = bannerConfigYupSchema.validateSync(parsed, { abortEarly: false });
+    return {
+      ...banner,
+      summary: DOMPurify.sanitize(banner.summary),
+    };
+  } catch {
+    return null;
```

Comment:
Can you do at least console log that it failed here?

Reply:
resolved. thank you.:-).